### PR TITLE
Rename Dummy to PickFirst load balancer

### PR DESCRIPTION
**Pull request by lukaszx0**
Renames `DummyLoadBalancer` to `PickFirstLoadBalancer` to better reflect what it actually does.

### CODE REVIEW

**Comment by ejona86** [comment]
@lukaszx0 LGTM.

**Comment by ejona86** [comment]
I'll let @zhangkun83 review as well.

**Comment by zhangkun83** [comment]
LGTM

**Comment by lukaszx0** [comment]
Pushed style fix, build should be green now.

File: core/src/test/java/io/grpc/util/RoundRobinLoadBalancerTest.java [hunk 1]
```diff
@@ -66,7 +67,7 @@
 import java.util.ArrayList;
 import java.util.List;
 
-/** Unit test for {@link DummyLoadBalancerFactory}. */
+/** Unit test for {@link PickFirstBalancerFactory}. */
```
Comment (zhangkun83) [inline core/src/test/java/io/grpc/util/RoundRobinLoadBalancerTest.java]: RoundRobinLoadBalancerFactory

Comment (lukaszx0) [inline core/src/test/java/io/grpc/util/RoundRobinLoadBalancerTest.java]: Ah, thanks. Fixed.

**Comment by zhangkun83** [comment]
Cherry-picked 577164c42f64fc84e1c949d2c4be9c199462e43c

**Pull request closed without merging**